dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/ControllerActionProducesResponseType/ControllerActionProducesResponseTypeAnalyzer.cs

```diff
@@ -82,7 +82,9 @@ private static void AnalyzeMethodDeclaration(SyntaxNodeAnalysisContext nodeAnaly
                         name.Equals("ProducesResponseType")
                 );
 
-                if (!hasProducesResponseType)
+                var returnType = methodDeclarationSyntax.ReturnType.ToString();
+                
+                if (!hasProducesResponseType && !returnType.Contains("Results"))
```

Comment:
can we use a more robust check in case the return type happens to be something containing `Results` but isn't typed results? Perhaps a namespace check?

Reply:
@navodwickra1 looked into this, it looks like we can check the namespace like below:
`nodeAnalysisContext.SemanticModel.GetTypeInfo(methodDeclarationSyntax.ReturnType).Type.ContainingNamespace` - could we give that a go?

Reply:
Any update on better checking for this type? Agreed that just checking if the name contains "Results" is not very robust. We will also want to use this change if possible in the other analyzers.

Reply:
We can make the check better, however we wouldn't be able to have unit tests for it because Navod's not sure (not am I) how we'd get that code to compile, which impacts our ability to check its namespace. If we can make the code compile, then we can improve this